src/App.tsx

```diff
@@ -1,26 +1,416 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  USER_ID,
+  getTodos,
+  createTodos,
+  deleteTodos,
+  updateTodos,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterStatus } from './types/FilterStatus';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoList } from './components/TodoList';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState('');
+  const [filterStatus, setFilterStatus] = useState(FilterStatus.All);
+
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [isAddingTodo, setIsAddingTodo] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [deletingTodoIds, setDeletingTodoIds] = useState<number[]>([]);
+
+  const [updatingTodoIds, setUpdatingTodoIds] = useState<number[]>([]);
+
+  const newTodoFieldRef = React.useRef<HTMLInputElement>(null);
+
+  const [editingTodoId, setEditingTodoId] = useState<number | undefined>(
+    undefined,
+  );
+
+  const [updatedTodoId, setUpdatedTodoId] = useState<number | undefined>(
```

Comment:
I strongly believe that u can handle this taks without those two states
use filter or smth else
let's try to get rid them